test/benchmarks/src/main/scala/scala/collection/immutable/RedBlackTreeBenchmark.scala

```diff
@@ -0,0 +1,67 @@
+package scala.collection.immutable
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra._
+
+import scala.util.Random
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(2)
+@Threads(1)
+@Warmup(iterations = 30)
+@Measurement(iterations = 30)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class RedBlackTreeBenchmark {
+
+  @Param(Array("0", "1", "10", "100", "1000", "10000"))
+  var size: Int = _
+
+  var nums: Range = _
+  val rnd = new Random(0)
+  var set1: TreeSet[Int] = _
+  var perm: Array[Int] = _ // repeatably pseudo-random permutation
+
+  @Setup(Level.Trial) def init: Unit = {
+    nums = 1 to size
+    set1 = TreeSet.from(nums)
+    perm = new Array[Int](size)
+    val rem = scala.collection.mutable.ArrayBuffer.from(nums)
+    perm = Array.fill(size)(rem.remove(rnd.nextInt(rem.size)))
+    assert(rem.size == 0)
+    assert(perm.sum == nums.sum)
+  }
+
+  @Benchmark
+  def build(bh: Blackhole): Unit =
+    bh.consume(TreeSet.from(nums))
+
+  @Benchmark
+  def buildRandom(bh: Blackhole): Unit =
+    bh.consume(TreeSet.from(perm))
+
+  @Benchmark
+  def iterator(bh: Blackhole): Unit = {
+    val it = set1.iterator
+    var res = 0
+    while(it.hasNext)
+      res += it.next()
```

Comment:
I'd be inclined to structure the test without any auto-unboxing, e.g. by using a `TreeSet[String]` and just feeding the strings into the `Blackhole` inside the loop. I don't think it will change the results of this benchmark, but it reduces the cognitive burden :)